src/main/java/io/supertokens/webserver/api/bulkimport/BulkImportAPI.java

```diff
@@ -0,0 +1,209 @@
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.bulkimport;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import io.supertokens.Main;
+import io.supertokens.bulkimport.BulkImport;
+import io.supertokens.bulkimport.BulkImportUserPaginationContainer;
+import io.supertokens.bulkimport.BulkImportUserPaginationToken;
+import io.supertokens.bulkimport.BulkImportUserUtils;
+import io.supertokens.multitenancy.exception.BadPermissionException;
+import io.supertokens.output.Logging;
+import io.supertokens.pluginInterface.bulkimport.BulkImportStorage.BULK_IMPORT_USER_STATUS;
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.StorageUtils;
+import io.supertokens.pluginInterface.bulkimport.BulkImportUser;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.utils.Utils;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class BulkImportAPI extends WebserverAPI {
+    public BulkImportAPI(Main main) {
+        super(main, "");
+    }
+
+    @Override
+    public String getPath() {
+        return "/bulk-import/users";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        // API is app specific
+
+        if (StorageLayer.isInMemDb(main)) {
+            throw new ServletException(new BadRequestException("This API is not supported in the in-memory database."));
+        }
+
+        String statusString = InputParser.getQueryParamOrThrowError(req, "status", true);
+        String paginationToken = InputParser.getQueryParamOrThrowError(req, "paginationToken", true);
+        Integer limit = InputParser.getIntQueryParamOrThrowError(req, "limit", true);
+
+        if (limit != null) {
+            if (limit > BulkImport.GET_USERS_PAGINATION_MAX_LIMIT) {
+                throw new ServletException(
```

Comment:
Are u aware of typical post body size if say we are adding 10000 users? it would be a useful information to be able to set max post body size in the nginx for example.

Reply:
well, 10k randomized users with multiple LMs and so on, are around 7.5MB (written to a json file). I don't know if that counts as typical or not.

Reply:
we just need to notify the SaaS team as to how much max post body size should be set for this endpoint specifically.

Reply:
well, I don't have a better number. It's around 7.5MB